src/anyp/TrafficMode.h

```diff
@@ -58,27 +67,99 @@ class TrafficMode
      *  - URL translation from relative to absolute form
      *  - Same-Origin verification is mandatory
      *  - destination pinning is recommended
-     *  - authentication prohibited
+     *  - Squid authentication prohibited
      */
     bool tproxyIntercept = false;
+};
+
+/**
+ * Set of 'mode' flags defining types of traffic which can be received.
+ *
+ * Use to determine the processing steps which need to be applied
+ * to this traffic under any special circumstances which may apply.
+ */
+class TrafficMode
+{
+public:
+    /// This port handles traffic that has been intercepted prior to being delivered
+    /// to the TCP client of the accepted connection and/or to us. This port mode
+    /// alone does not imply that the client of the accepted TCP connection was not
+    /// connecting directly to this port (since commit 151ba0d).
+    bool interceptedSomewhere() const { return flags_.natIntercept || flags_.tproxyIntercept || flags_.proxySurrogateHttps; }
```

Comment:
Note that though this PR adds proxySurrogateHttps flag, it is currently unused - i.e., has a default 'false' value which does not affect anything. The actual limitation of proxySurrogateHttps usage is determined by PR832,
which binds proxySurrogateHttps (i.e.,PROXY protocol on https_port) to one case only - TLS traffic intercepted somewhere and directed to this Squid instance, preceded by the PROXY protocol header. Having this proxySurrogateHttps definition in mind, I hope you agree that 'interceptedSomewhere()' covers this use quite well.

Reply:
> Amos: You missed out proxySurrogateHttp

If I interpret Eduard's response correctly, his answer is, essentially, "No, I did not. proxySurrogateHttp does not imply interceptedSomewhere()".

I agree with (my interpretation of) that answer: "http_port require-proxy-header" does not imply that the traffic was intercepted somewhere. For example, I can configure a client to use haproxy (or equivalent) as a forward proxy and then configure haproxy to use Squid as a forward proxy (while adding the PROXY protocol header). The resulting traffic will not be intercepted at any point so interceptedSomewhere() will return false while proxySurrogateHttp will be true. No implication.

> IMO this is too vague a condition

The "intercepted at some point" condition feels well defined to me. Can you give an example that illustrates its alleged vagueness?

Reply:
@yadij, please revisit this change request and review the current definition of interceptedSomewhere(). I believe recent changes make that method implementation clearer. Since many of your other change requests depend on the this method, let's try to get an agreement here first. By default, a review request applies to the whole PR. Due to numerous inter-dependencies and misunderstandings in this PR change requests, it would be very difficult to make progress on many fronts IMO. Thus, the review request I am going to make after posting this comment applies to a single interceptedSomewhere() method.

If it all possible, please ignore the spelling of the method names for now and focus on the combination of flags they return. Once we reach an agreement on flags, we will have a much better chance of finding a good name for that combination.

Reply:
recent changes make it clearer. resolving this thread.